mrbeam_ledstrips/server.py

```diff
@@ -187,17 +187,25 @@ def start(self):
 		self.animation.daemon = True
 		self.animation.name = "StateAnimations"
 		self.animation.start()
-		self._socket_monitor(self.server_address, callback=self.on_state_change)
+		self._socket_monitor(self.server_address, callback=self.parse_socket_msg)
 
-	def on_state_change(self, state):
+	def parse_socket_msg(self, command):
 		response = "ERRROR"
-		if (state in ('info', '?')):
+		if command in ('info', '?'):
 			info = self.get_info()
 			self.logger.info(info)
-			response = info
+			return info
+		elif command.startswith("set:"):
+			available_settings = list(chain(SETTINGS.values()))
+			_split = command.split(":")
+			if len(_split) <= 2:
+				return "ERROR : Need to specifiy set:SETTING:VAL where SETTING is \{{}\}".format(",".join(available_settings))
+			elif _split[1] not in available_settings:
+				return "ERROR, {} not in available settings \\{{}\\}".format(_split[1], available_settings)
+			setting = _split[1]
+			return self.leds.set_setting(_split[1], _split[2:])
 		else:
-			response = self.leds.change_state(state)
-		return response
+			return self.leds.change_state(command)
```

Comment:
so, what exactly the command we expect in this case? don't we need a sanity check here as well ?

Usually else is for unsupported commands

Reply:
more specifically:
I moved the `set_settings()` logic out of the `change_state()` function.